### ISSUE TITLE
Xcode 7.1 - Include of non-modular header inside framework

### ISSUE THREAD

**Issue by retsohuang** [ReactiveX/RxSwift#210]
Same issue as CocoaPods/CocoaPods#4420, any ideas?

**Comment by kzaher** [comment]
Hi @retsohuang ,

first news to me.

I've rerun our deployment integration tests and they all pass. I've also created a new project and it seems to be working for me.

This is Podfile

```
use_frameworks!
pod 'RxSwift', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
pod 'RxCocoa', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
pod 'RxBlocking', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
```

Cocoapods 0.39
Xcode Version 7.1 (7B91b)
OSX 10.11.1 (15B42)
created new project for iOS platform from Single page template app.

Please let me know does this work, and is there anything we can do to help on project level.

Kruno

**Comment by retsohuang** [comment]
I'll test that later, but my project is existing project written in objc and contain Swift code, maybe is the problem?

**Comment by kzaher** [comment]
Oh, that is probably it.

¯\_(ツ)_/¯

**Comment by retsohuang** [comment]
I tried to reproduce this issue by create project and language choose objc, but I passed, Am I missing any setting in existing project?

**Comment by kzaher** [comment]
Hm,

can you paste me your `Podfile`?

Frankly, I have no idea what is wrong :)

Check for .h files in your framework (RxSwift/RxCocoa) pods project. Try to play with 

<img width="728" alt="screen shot 2015-10-26 at 11 06 41" src="https://cloud.githubusercontent.com/assets/1641148/10726240/b5c7d9e2-7bd1-11e5-84c2-68a3ba8d8218.png">

If you could send me something that fails for you I could take a closer look. 

Without repro locally and with a lot of other people having the same issue, the chances of me solving this are slim :(

**Comment by retsohuang** [comment]
Heres my podfile
```
platform :ios, '8.0'
inhibit_all_warnings!
use_frameworks!

source 'https://github.com/CocoaPods/Specs.git'

pod 'AMPopTip'

# Swift Library
pod 'AlamofireImage'
pod 'Bond'
pod 'Moya/RxSwift'
pod 'RxSwift', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
pod 'RxCocoa', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
pod 'RxBlocking', :git => 'git@github.com:ReactiveX/RxSwift.git', :branch => '2.0.0-beta.1'
pod 'Spring', :git => 'https://github.com/MengTo/Spring.git', :branch => 'swift2'
pod 'SwiftyJSON'
pod 'SwiftyUserDefaults'
pod 'PKHUD', :git => "https://github.com/retsohuang/PKHUD.git", :branch => 'feature/swift2.0'
pod 'SwiftColors', :git => 'https://github.com/thii/SwiftColors.git', :branch => 'swift-2.0'
pod 'TBMultiAppearanceButton'
pod 'Eureka'
pod 'SnapKit', :git => 'https://github.com/SnapKit/SnapKit.git', :branch => 'swift-2.0'
pod 'RSBarcodes_Swift'
pod 'FloatRatingView', :git => 'https://github.com/strekfus/FloatRatingView.git'
pod 'RxAlamofire'
pod 'JTSImageViewController'
pod 'UIView+draggable'
```

And the screenshot
<img width="1438" alt="screen shot 2015-10-26 at 6 21 35 pm" src="https://cloud.githubusercontent.com/assets/785920/10726575/77384e22-7c0e-11e5-8215-f613bb8f3ba5.png">

**Comment by kzaher** [comment]
Hi @retsohuang ,

I've made a [mixed test project](https://www.dropbox.com/s/q7a610nr8ai2225/Mixed.zip?dl=0) that uses the same Podfile and it looks like it's working.

Can you confirm it's working in your environment?

**Comment by retsohuang** [comment]
Hi @kzaher 
I've done the same thing before and also working too, I have no idea why my exist project (not created via Xcode 7.1) get this error.

**Comment by zhigang1992** [comment]
https://github.com/Quick/Quick/issues/406#issuecomment-151194694

**Comment by retsohuang** [comment]
Hi @zhigang1992 

Sorry it's not working.

**Comment by kzaher** [comment]
Hi @retsohuang ,

Have you maybe tried updating your CocoaPods? What is the version of CocoaPods you are using?

**Comment by retsohuang** [comment]
CocoaPods: 0.39.0
Xcode version: 7.1

**Comment by kzaher** [comment]
Sorry @retsohuang , I'm not sure I know what else can we do if it's working for a fresh project. There is probably something misconfigured in your project. Hard to debug remotely.

¯\_(ツ)_/¯ 

If a lot of people will have this issue, or if we'll have some new insight, we can respond to this thread.

The only solution that I know of now is just recreating project file, and that should be working from your response.

Sorry :)

**Comment by hilen** [comment]
Same error, my pod version is ```1.0.0.beta.2```
pod file is 
```
    pod 'RxSwift',    '~> 2.0'
    pod 'RxCocoa',    '~> 2.0'
    pod 'RxBlocking', '~> 2.0'
```

How to fix it?

**Comment by callmebill** [comment]
Hi @retsohuang 
I just remove  RxSwift and RxCocoa  . 
    #pod 'RxSwift',    '~> 3.0'
    #pod 'RxCocoa',    '~> 3.0'
Then build successed . 
Add Them  .
    pod 'RxSwift',    '~> 3.0'
    pod 'RxCocoa',    '~> 3.0'
Then build successed  with RxSwift and RxCocoa ! It work for me .

**Comment by devRizvan13** [comment]
@retsohuang @huangzhigangxg 
I have the same problem. 
Tried all of the above solutions.
Added RxSwift library via CocoaPod. The project is written in Swift 2.3.
pod 'RxSwift', '~> 2.6.1'
pod 'RxCocoa', '~> 2.6.1'
When compiling, there was an error #import <RxCocoa / RxCocoa.h> Enabling the non-modular header inside the frame module 'RxCocoa.Swift'
I deleted this line #import <RxCocoa / RxCocoa.h> in the RxCocoa-Swift.h file, when the project was started on the simulator or under consideration, the error stopped coming out. But when I try to archive a project to send to iTunesConnect, it drops back. And the line, I deleted #import <RxCocoa / RxCocoa.h>, will appear again.
